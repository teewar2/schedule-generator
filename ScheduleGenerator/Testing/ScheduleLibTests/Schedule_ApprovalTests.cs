using System;
using System.Linq;
using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;
using static Infrastructure.SheetConstants;
using static Domain.DomainExtensions;

namespace Testing.ScheduleLibTests
{
    [TestFixture]
    [Explicit]
    [UseReporter(typeof(DiffReporter))]
    public class Schedule_ApprovalTests
    {
        [Test]
        public void CheckMeetingsPlaced_Approval()
        {
            var solver = GetSolver(SpringConfig, Repository);
            var schedule = solver
                .GetSolution(new(0, 0, 0, 0, 100))
                .Schedule;

            var placedMeetings = schedule
                .Meetings
                .Select(m => $"{m.Discipline} {m.MeetingType} {m.Teacher}")
                .OrderBy(m => m)
                .ToList();
            var notePlacedMeetings = schedule
                .NotUsedMeetings
                .Select(m => $"{m.Discipline} {m.MeetingType} {m.Teacher}")
                .OrderBy(m => m)
                .ToList();
            var unifiedString =
                $"Placed:{placedMeetings.Count} Left:{schedule.NotUsedMeetings.Count}, Placed:{Environment.NewLine}" +
                $"{string.Join(Environment.NewLine, placedMeetings)}{Environment.NewLine}" +
                $"Not placed:{Environment.NewLine}" +
                $"{string.Join(Environment.NewLine, notePlacedMeetings)}{Environment.NewLine}";
            Approvals.Verify(unifiedString);
        }
    }
}