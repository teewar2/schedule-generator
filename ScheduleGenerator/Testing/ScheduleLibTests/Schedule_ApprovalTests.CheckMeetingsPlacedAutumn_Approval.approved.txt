Placed:116 Left:0
Placed:
Алгем Lecture Александр Гейн
Алгем Lecture Александр Гейн
Алгем Lecture Михаил Волков
Алгем Lecture Михаил Волков
Алгем Seminar Александр Белов
Алгем Seminar Александр Белов
Алгем Seminar Александр Бондарь
Алгем Seminar Александр Бондарь
Алгем Seminar Валентина Щербакова
Алгем Seminar Валентина Щербакова
Алгем Seminar Василий Климин
Алгем Seminar Василий Климин
Алгем Seminar Дмитрий Косолобов
Алгем Seminar Дмитрий Косолобов
Алгем Seminar Инна Михайлова
Алгем Seminar Инна Михайлова
Алгем Seminar Олег Расин
Алгем Seminar Олег Расин
Алгем Seminar Павел Гейн
Алгем Seminar Павел Гейн
ДМ Lecture Арсений Шур
ДМ Lecture Арсений Шур
ДМ Seminar Арсений Шур
ДМ Seminar Инна Михайлова
ДМ Seminar Никита Сивухин
ДМ Seminar Олег Расин
ИнЯз Seminar Кафедра английского
Матанализ Lecture Иван Симонов
Матанализ Lecture Иван Симонов
Матанализ Lecture Симонов Иван
Матанализ Lecture Симонов Иван
Матанализ Lecture Ульяна Алексеева
Матанализ Lecture Ульяна Алексеева
Матанализ Lecture Филатова Мария
Матанализ Lecture Филатова Мария
Матанализ Seminar Александр Бондарь
Матанализ Seminar Александр Бондарь
Матанализ Seminar Волков Алексей
Матанализ Seminar Волков Алексей
Матанализ Seminar Волков Алексей
Матанализ Seminar Волков Алексей
Матанализ Seminar Липин Антон
Матанализ Seminar Липин Антон
Матанализ Seminar Максим Першаков
Матанализ Seminar Максим Першаков
Матанализ Seminar Никифорова Татьяна
Матанализ Seminar Никифорова Татьяна
Матанализ Seminar Никифорова Татьяна
Матанализ Seminar Никифорова Татьяна
Матанализ Seminar Симонов Иван
Матанализ Seminar Симонов Иван
Матанализ Seminar Торгашева Настя
Матанализ Seminar Торгашева Настя
Матанализ Seminar Ульяна Алексеева
Матанализ Seminar Ульяна Алексеева
Матанализ Seminar Филатова Мария
Матанализ Seminar Филатова Мария
Матанализ Seminar Ямковой Дмитрий
Матанализ Seminar Ямковой Дмитрий
ООП ComputerLab Александр Сорокин
ООП ComputerLab Александр Сорокин
ООП ComputerLab Алексей Политов
ООП ComputerLab Даниил Айзенштейн
ООП ComputerLab Даниил Айзенштейн
ООП ComputerLab Игорь Белоусов
ООП ComputerLab Михаил Плисковский
ООП ComputerLab Павел Егоров
ОП ComputerLab Алексей Политов
ОП ComputerLab Андрей Петряшов
ОП ComputerLab Антон Федоров
ОП ComputerLab Артем Маценко
ОП ComputerLab Балашов Михаил Александрович
ОП ComputerLab Голубев Дмитрий Сергеевич
ОП ComputerLab Ольга Романовская
ОП ComputerLab Хапов Кирилл
ОП Seminar Алексей Политов
ОП Seminar Андрей Петряшов
ОП Seminar Антон Федоров
ОП Seminar Артем Маценко
ОП Seminar Балашов Михаил Александрович
ОП Seminar Голубев Дмитрий Сергеевич
ОП Seminar Ольга Романовская
ОП Seminar Хапов Кирилл
ОПД Lecture Иван Домашних
ОПД Lecture Иван Домашних
ОПД Lecture Иван Домашних
ОПД Lecture Иван Домашних
ОС ComputerLab Бабычев
ОС ComputerLab Кошелев Гриша 
ОС ComputerLab Лила Вова
ОС ComputerLab Щукина
ОС Lecture Вячеслав Копейцев
Психология Lecture Владимир Савельев
Психология Lecture Владимир Савельев
Психология Seminar Владимир Савельев
Психология Seminar Владимир Савельев
ПЭК Seminar Алексей Кислов
Сети и протоколы ComputerLab Александр Берсенев
Сети и протоколы ComputerLab Андрей Гейн
Сети и протоколы ComputerLab Арина Яровая
Сети и протоколы ComputerLab Наталья Неустроева
Сети и протоколы Lecture Наталья Неустроева
Теорвер Lecture Дмитрий Хлопин
Теорвер Lecture Дмитрий Хлопин
Теорвер Seminar Александр Колиниченко
Теорвер Seminar Александр Колиниченко
Теорвер Seminar Илья Цуп 
Теорвер Seminar Семён Трифочкин
Физкультура Seminar Кафедра физкультуры
Физкультура Seminar Кафедра физкультуры
Физкультура Seminar Кафедра физкультуры
Физкультура Seminar Кафедра физкультуры
Язык Python ComputerLab Анкудинов
Язык Python ComputerLab Берсенёв
Язык Python ComputerLab Зверев
Язык Python ComputerLab Мясников
Not placed:

